osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -430,10 +440,11 @@ protected override async Task CleanUpState(MultiplayerClientState state)
         private async Task setNewHost(MultiplayerRoom room, MultiplayerRoomUser newHost)
         {
             room.Host = newHost;
+            await Clients.Group(GetGroupId(room.RoomID)).HostChanged(newHost.UserID);
 
-            await UpdateDatabaseHost(room);
 
-            await Clients.Group(GetGroupId(room.RoomID)).HostChanged(newHost.UserID);
+            // don't care too much about failures here (is only for display), so run after local state is updated.
+            await UpdateDatabaseHost(room);
```

Comment:
still not 100% safe to do that i don't think. if the usage in `leaveRoom()` throws, it won't notify clients that the user has left.

probably safest to mirror `UpdateDatabaseParticipants()` for now.